src/shared/bus-unit-util.c

```diff
@@ -2265,6 +2268,33 @@ static int bus_append_execute_property(sd_bus_message *m, const char *field, con
                 return 1;
         }
 
+        if (STR_IN_SET(field, "StateDirectoryQuota", "CacheDirectoryQuota", "LogsDirectoryQuota")) {
+                uint64_t quota_absolute = UINT64_MAX;
+                uint32_t quota_scale = UINT32_MAX;
+                int quota_enforce = false;
+
+                if (!isempty(eq) && !streq(eq, "off")) {
+                        r = parse_permyriad(eq);
+                        if (r < 0) {
+                                uint64_t bytes = UINT64_MAX;
+                                r = parse_size(eq, 1024, &bytes);
+                                if (r < 0)
+                                        return bus_log_create_error(r);
```

Comment:
wrong logging function. that one is for failures when prepping dbus messages. just use log_error_errno() and write a proper error string